### ISSUE TITLE
LambdaLift lifts methods that access private fields too far

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#514]
`Contexts.Context.printCreationTraces` has a lambda inside, that accesses a private accessor `creationTrace` of `Contexts.Context`.
LambdaLift lifts it out to a static method in `Contexts`, without setting a `NotJavaPrivate` on the private accessor of the `creationTrace`.

Manifests itself with such error:
```
Exception in thread "main" java.lang.VerifyError: Bad invokespecial instruction: current class isn't assignable to reference class.
Exception Details:
  Location:
    dotty/tools/dotc/core/Contexts$.$anonfun$printCreationTraces$2(Ldotty/tools/dotc/core/Contexts$$Context;)V @50: invokespecial
  Reason:
    Error exists in the bytecode
```

**Comment by DarkDimius** [comment]
@odersky If you decide that you can lift method to be static, why make it static on top-level class, instead of making it static on the class that was the original owner?

**Comment by odersky** [comment]
Addressed by #517
